Automattic/jetpack
# Social: Change the connections endpoint to connection-test-results

After merging #26216, we discovered that Jetpack Social was not using the same API endpoint to get the connection data in the editor. The one used by Jetpack and WPCOM already provided information about the health of the connection.

#### Changes proposed in this Pull Request:
This change reverts the addition of the `is_healthy` property in the `/publicize/connections` endpoint and updates Jetpack Social to use the `/publicize/connection-test-results` endpoint.



#### Other information:


#### Jetpack product discussion


p1663318602723539-slack-C02JJ910CNL

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:
* With this PR and Jetpack Social active
* Load the block editor and toggle the sidebar for social
* In the network tab of developer tools you should see a request to `/jetpack/v4/publicize/connection-test-results`
* If you apply the associated patch and sandbox the WPCOM API, you should see that `is_healthy` is no longer returned by that endpoint.
